vpcinfo.go

```diff
@@ -122,27 +157,30 @@ func whereAmI() (Platform, error) {
 		s := string(b)
 		switch {
 		case strings.HasPrefix(s, "EC2"), strings.HasPrefix(s, "ec2"):
-			return aws{}, nil
+			zone, err := zoneFromSubnets(subnets)
+			if err == nil {
+				return aws{zone}, nil
```

Comment:
I feel like we shouldn't use errors to report "no zone found", it causes us to silence errors that would occur when calling `net.InterfaceAddrs()`.

How about a change like this?
```go
zone, err := zoneFromSubnets(subnets)
if err != nil {
    return nil, err
}
if zone != "" {
    return aws{zone}, nil
}
```

Reply:
Done.